analyzers/tests/SonarAnalyzer.TestFramework/Common/SourceGeneratorProvider.cs

```diff
@@ -26,13 +26,33 @@ namespace SonarAnalyzer.TestFramework.Common;
 public static class SourceGeneratorProvider
 {
     private static readonly string RazorSourceGeneratorPath =
-        Path.Combine(Path.GetDirectoryName(typeof(SourceGeneratorProvider).Assembly.Location), "Dependencies", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
+        Path.Combine(LatestSdkFolder(), "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", "Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll");
 
     public static AnalyzerFileReference[] SourceGenerators { get; } =
     [
         new(RazorSourceGeneratorPath, new AssemblyLoader())
     ];
 
+    public static string LatestSdkFolder()
+    {
+        var objectAssembly = typeof(object).Assembly;
+        var objectAssemblyDirectory = Directory.GetParent(objectAssembly.Location);    // C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.4
+        var dotnetDirectory = objectAssemblyDirectory.Parent.Parent.Parent;            // C:\Program Files\dotnet
+        var sdkDirectory = Path.Combine(dotnetDirectory.FullName, "sdk");              // C:\Program Files\dotnet\sdk
+
+        if (!Directory.Exists(sdkDirectory))
+        {
+            throw new NotSupportedException($"The directory '{sdkDirectory}' does not exist. " +
+                $"This may be because you are not using .NET Core. " +
+                $"Please note that Razor analysis is only supported when using .NET Core.");
+        }
+        return Directory.GetDirectories(sdkDirectory, $"{objectAssembly.GetName().Version.Major}.*", SearchOption.TopDirectoryOnly)
```

Comment:
Didn't we say that we'll use the same version, as the current SDK? And that will guarantee that such file exists? 
The `OrDefault` can be removed, simplifying the method.

Reply:
That's what the implementation is doing 🤔 we are using the same `major` version of `typeof(object).Assembly`, picking the highest `feature`-`patch`. I do agree that OrDefault can be removed because we should always have at least one tho

Reply:
No, it uses the oldest (lowest possible version) installed version of the same major version as the projects targets.

It should take the exact version as seen in objectAssembly.Location path and use that. Because that one is the latest installed due to global.json file.

Reply:
I don't think this is what the code does.
Can you please try to debug `LatestSdkFolder_ReturnLatest` on your machine and see what sdk is returned from `LatestSdkFolder()` call?
In my case, I have the following:
* `3.1.426`
* `5.0.408`
* `6.0.421`
* `7.0.408`
* `8.0.104`
* `8.0.204`

And `LatestSdkFolder()` returns correctly: `C:\Program Files\dotnet\sdk\8.0.204`.

Regarding the point of using the exact version as in objectAssembly.Location, it is technically feasible and would yield identical results. However, we will depend on the definitions in the global.json file

Reply:
Hmm, for whatever reason, I have 8.0.4 as object 8.0.204 as SDK, so it would not work if we take the exact version :/ 

I overlooked the OrderByDescending. I'd prefer to have OrderBy().Last()